hawkbit-ddi-api/src/main/java/org/eclipse/hawkbit/ddi/json/model/DdiStatus.java

```diff
@@ -31,6 +32,7 @@
     @Valid
     private final DdiResult result;
 
+    @Size(min = 0, max = 50)
```

Comment:
This is explained to some extent in the first commit that is part of this pull request: 

" Even though the number of ActionStatus entries is limited to 1000 per action using QuotaManagement, there is no limit on the number of messages that can be sent as part of a single ActionStatus. This allows a controller to potentially send large number of messages for a single action status. Hence, limit the number of messages allowed to 50 using the javax.validation.constraints.Size within DdiStatus."

Since there is no limitation of messages per action status, currently it is possible that the last action status reports for example 200 or even larger number of messages. Besides the potential of flooding the database, we may only get be able to retrieve part of the last feedback.

This is because we just get a flat list of messages based on the message count and limit the maximum possible count to 100 messages for retrieval. With limit of 50 above, we are can get back at least 2 complete set of action status messages.